gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -183,8 +203,12 @@
   }
 
   private void createEmptyWorkUnitsForSkippedPartitions(Map<String, List<WorkUnit>> workUnits,
-      Map<String, State> topicSpecificStateMap) {
+      Map<String, State> topicSpecificStateMap, SourceState state) {
 
+    if(!this.doneGettingAllPreviousOffsets.get()){
```

Comment:
Why do you need this check? Isn't this check already in `getAllPreviousOffserts(state)`?